dev/com.ibm.ws.concurrent.persistent_fat_multiple/fat/src/com/ibm/ws/concurrent/persistent/fat/multiple/DatabaseContainerFactory.java

```diff
@@ -0,0 +1,126 @@
+package com.ibm.ws.concurrent.persistent.fat.multiple;
+
+import java.time.Duration;
+
+import org.testcontainers.containers.Db2Container;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.MSSQLServerContainer;
+import org.testcontainers.containers.OracleContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.output.OutputFrame;
+
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.custom.junit.runner.FATRunner;
+
+/**
+ * This is a factory class that creates a database test container.
+ * The test container returned will be based on the {fat.bucket.db.type} 
+ * system property. 
+ * 
+ * The {fat.bucket.db.type} property is set to different databases
+ * by our test infrastructure when a fat-suite has the property
+ * {fat.test.databases} set to true.
+ * 
+ * <br> Container Information: <br>
+ * DERBY: Uses a derby no-op test container <br>
+ * DB2: Uses <a href="https://hub.docker.com/r/ibmcom/db2">Offical DB2 Container</a> <br>
+ * Oracle: TODO replace this container with the official oracle-xe container if/when it is available without a license. <br>
+ * Postgre: Uses <a href="https://hub.docker.com/_/postgres">Offical Postgres Container</a> <br>
+ * MS SQL Server: Uses <a href="https://hub.docker.com/_/microsoft-mssql-server">Offical Microsoft SQL Container</a> <br>
+ * 
+ * TODO: Since this factory class is general it should be moved to fattest simplicity in the future
+ */
+public class DatabaseContainerFactory {	
+	private static final Class<DatabaseContainerFactory> c = DatabaseContainerFactory.class;
+	private static JdbcDatabaseContainer<?> container;
+	private static DatabaseContainerType dbContainerType;
+	
+	/**
+	 * @see #create(String databaseName)
+	 * 
+	 * Uses a default database name 'test'
+	 */
+	public static JdbcDatabaseContainer<?> create() {
```

Comment:
is this method ever used? I think we can just remove it

Reply:
I had left this here as a option later on if we have a test suite where we do not care what the name of the database is.  

And also as a general factory class I guess we shouldn't assume that it will be used specifically for database rotation.  I guess any test suite could use it to create a test-container if they wanted to.